RESPONDER: fix responder_idle_handler()

alexey-tikhonov opened the pull request:
Since timer fires every timeout/2, with condition '>' it could happen
responder was terminated only after 3*timeout/2 of being idle.
Changing condition to '>=' aligns actual behavior with man page.

RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=2097014

pbrezina commented:
Hi Alexey, can you please explain the problem in more detail in the commit message?

alexey-tikhonov commented:
Of course, even with this patch responder can run up to 3/2*timeout:
 - initial request / start: t0 - timer started
 - 2nd request: (t0 + 1) - idle counter reset
 - t0 + timeout/2 - time fires - condition not met
 - t0 + 2*timeout/2 - time fires - condition not met
 - t0 + 3*timeout/2 - time first - condition is met in (3*timeout/2 - 1) after 2nd request

To fully "fix" the issue we should either restart timer at every idle counter reset or make it fire very often. But imo it's not worth the effort. This trivial patch fixes most glaring (and common) case - sssd_ifp is activated to handle single request.

pbrezina reviewed (approved):
Thank you for the explanation, now it is more clear why this was needed.

aplopez reviewed (approved):
LGTM

thalman reviewed (approved):
LGTM, ACK

alexey-tikhonov commented:
Pushed PR: https://github.com/SSSD/sssd/pull/6277

* `master`
    * 473752e261d8d166b16f95fe4ddc8d320ae0faf4 - RESPONDER: fixed condition in responder_idle_handler()
* `sssd-1-16`
    * f37e15e9134ff22d03c4997fd1c76754b67d3539 - RESPONDER: fixed condition in responder_idle_handler()
